src/app/order/OrderStatus.tsx

```diff
@@ -17,21 +17,17 @@ const OrderStatus: FunctionComponent<OrderStatusProps> = ({
     supportPhoneNumber,
 }) => {
 
-    const getMandateProvider = useCallback(() => {
-        return order?.payments?.[0].description;
-    }, [order]);
+    let mandateProvider = '';
+    let mandateMethod = '';
+    let orderMandate;
 
-    const getMandateTextId = useCallback(() => {
-        const Mandates = [
-            { method: 'Stripe (SEPA)', value: 'sepa_link_text' },
-            { method: 'OXXO (via Checkout.com)', value: 'oxxo_link_text' },
-            { method: 'Boleto Bancário (via Checkout.com)', value: 'boleto_link_text' },
-        ];
+    const mandatePayment = order?.payments?.find((payment: GatewayOrderPayment | GiftCertificateOrderPayment)  => 'mandate' in  payment) as GatewayOrderPayment;
```

Comment:
Using `as` is an anti-pattern that we should try and avoid. Casting as `GatewayOrderPayment` is also not quite specific enough for our use cases, which is why we still have so many existence checks for certain attributes.

We can ensure our required entities more cleanly with a Type Guard, something like:

```tsx
type PaymentWithMandate = GatewayOrderPayment & Required<Pick<GatewayOrderPayment, 'mandate' | 'methodId'>>;

const isPaymentWithMandate = (payment: GatewayOrderPayment | GiftCertificateOrderPayment): payment is PaymentWithMandate =>
    !!payment.methodId && !!payment.mandate;
```

Where `PaymentWithMandate` is a sub-type of `Payment` which conforms to shape of data that our component requires. 

With that we can remove the `let`s (a bit of a code smell in functional component) and instead filter the payments upfront:

```tsx
const paymentsWithMandates = order?.payments?.filter(isPaymentWithMandate) || [];
```

Meaningly we can safely cycle over them in the JSX (with all values ensured, except `url`):

```tsx
{ paymentsWithMandates.map(payment => {
    if (payment.mandate.url) {
            // render it with a link... 
        );
    }

    // else render it as just text...
}) }
```

It might also be easier to read if the two different mandate JSX blocks are turned into their own Functional Components 🍹 

(Ideally we'd move all this data shaping out of this component to the source, but that'd be too much refactoring for this ticket)